libvncserver/rfbserver.c

```diff
@@ -2013,6 +2028,170 @@ rfbBool rfbProcessFileTransfer(rfbClientPtr cl, uint8_t contentType, uint8_t con
     return FALSE;
 }
 
+#ifdef LIBVNCSERVER_HAVE_LIBZ
+static rfbBool
+rfbSendExtendedClipboardCapability(rfbClientPtr cl) {
+    char buf[16] = {
+        0x03, 0x00, 0x00, 0x00,
+        0xFF, 0xFF, 0xFF, 0xF8, /* -8 */
+        0x17, 0x00, 0x00, 0x01, /* text, request, peek, provide */
+        0x00, 0x10, 0x00, 0x00, /* max size is 1MiB */
+    };
+    if (rfbWriteExact(cl, buf, sizeof(buf)) < 0) {
+        rfbLogPerror("rfbSendExtendedClipboardCapability: write");
+        rfbCloseClient(cl);
+        return FALSE;
+    }
+    rfbStatRecordMessageSent(cl, rfbServerCutText, 16, 16);
+    return TRUE;
+}
+
+static rfbBool
+rfbSendExtendedClipboardNotify(rfbClientPtr cl) {
+    char buf[16] = {
+        0x03, 0x00, 0x00, 0x00,
+        0xFF, 0xFF, 0xFF, 0xFC, /* -4 */
+        0x08, 0x00, 0x00, 0x01, /* only text */
+    };
+    if (rfbWriteExact(cl, buf, sizeof(buf)) < 0) {
+        rfbLogPerror("rfbSendExtendedClipboardNotify: write");
+        rfbCloseClient(cl);
+        return FALSE;
+    }
+    rfbStatRecordMessageSent(cl, rfbServerCutText, 12, 12);
```

Comment:
Me thinks 16 bytes were written?

Reply:
Sorry it's my typo. The buffer size should be 12 but it seems that extra four zero bytes happen to be a empty FramebufferUpdate so the stream will not be corrupted. I will fix it.